File: tier1/java/cd37d891/Report.java
import java.util.function.Function;

public final class Report {
    public static String build(String code) {
        Function<String, String> compose =
            value -> "SELECT status FROM orders WHERE code = ?";
        return apply(compose, code);
    }

    private static String apply(Function<String, String> step, String value) {
        return run(step.apply(value), value);
    }

    private static String run(String statement, String value) {
        return statement + "|" + value;
    }
}

File: tier1/csharp/6392e688/Report.cs
using System;

public static class Report
{
    public static string Build(string code)
    {
        Func<string, string> compose = value =>
            "SELECT status FROM orders WHERE code = @code";
        return Apply(compose, code);
    }

    private static string Apply(Func<string, string> step, string value) =>
        Run(step(value), value);

    private static string Run(string statement, string value) =>
        statement + "|" + value;
}
